src/main/java/org/teamresistance/frc/autonomous/AutoPingController.java

```diff
@@ -0,0 +1,32 @@
+package org.teamresistance.frc.autonomous;
+
+import org.teamresistance.frc.subsystem.Controller;
+import edu.wpi.first.wpilibj.PIDSourceType;
+import org.strongback.control.SoftwarePIDController;
+import org.teamresistance.frc.Pose;
+import org.teamresistance.frc.subsystem.Controller;
+import org.teamresistance.frc.util.SynchronousPID;
+
+
+/**
+ * Created by owner on 1/25/2017.
+ */
+public class AutoPingController implements Controller {
```

Comment:
Can you update `Controller` to be `Controller<Drive.Signal>`? Then you'll need to replace all mentions of `Object`to `Drive.Signal` in this class (or else the class won't be implementing the interface and the IDE will scream).

That's the gist of how Java's **generics** works. If you peek inside `Controller`, you'll see that `T` is being used as a placeholder type. When no `T` is specified, Java replaces all the `T`'s with the default `Object`. And so this Controller was actually a `Controller<Object>` before.

Making this a parameterized `Controller<Drive.Signal>` makes it a compatible input to the `Drive#setController` method. The Drive subsystem itself is a parameterized `ClosedLooping<Drive.Signal>` class. `Controller` and `ClosedLooping` are what we would call **generic** interfaces.